eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -2298,6 +2298,17 @@ default String makeString(String start, String separator, String end)
         return stringBuilder.toString();
     }
 
+    /**
+     * Returns a string representation of the collection, created by applying the function supplied to each element,
+     * with the elements separated by the specified separator and enclosed between the start and end strings.
+     *
+     * @return a string representation of the mapped collection
+     */
+    default String makeString(Function<? super T, String> function, String start, String separator, String end)
```

Comment:
I'm thinking it would be better to have the `function` return `Object` as the type and let `makeString` apply `String.valueOf()` as it does currently. I was originally thinking `? extends CharSequence` to allow for more "String-like" implementations, but there is no need to be restrictive here as `makeString` works with any object currently.

Reply:
done